### PR TITLE
Fix CMS edit goose save functionality

### PR DESCRIPTION
**Pull request by klau1011**
### [CLICKUP TICKET LINK](https://app.clickup.com/t/1e6ebrd)

Fixed edit goose functionality to update properly to API and state on save

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Manually tested to ensure field values are updated on save 
- [ ] Does not save on missing field 
- [ ] Normal flow of website

### CODE REVIEW

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:88]: Can we name this variable to something different (e.g., `newlyAddedImages`)? The current variable name of `files` isn't very descriptive

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:84]: We can probably delete these two lines since we did validation checks below

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:99]: Is this redundant (meaning we can remove it)? Because regardless if we are adding files or not, updating the state of `imagesUrls` using a map of `imageUrls` is basically setting it equal to itself, except in this case the structure of the state is actually being altered (this code makes `imageUrls` an array of strings, rather than an array of objects, which is what it was originally (see line 29 of this file) - although the state doesn't matter too much since it gets overriden once you open the form again, this should be avoided)

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:93]: Do we need to enter this block if `imagesUrls.length > 0`? Because we only want to upload new images, which is represented by `files`. `imageUrls` is for images that are already uploaded

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:102]: We can remove this if you alter the condition to only enter this if block if there are new images added

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:91]: What's the purpose of this array? Can we name it to something more descriptive, if needed?

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:125]: This logic works theoretically for the current `imageUrls2`. This is because if there are new images, then this array has all the new images. If there are no new images, then this array holds the previous images, which both pass the check of having enough images. However, if we have the logic where one variable holds the new images (e.g., `newlyAddedImages`) and another variable holds the current images that are stored in the database (e.g., `previousImages`), then we can do something along the lines of `(newlyAddedImages.length > 0 || previousImages.length > 0`), which makes more sense. You also have to take into account when a user deletes all the previous images, which is stored in `imagesToDelete`

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:135]: Can consider removing if the above logic is implemented

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:147]: We can probably delete this line too

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:173]: One interesting bug that I noticed was that the form wouldn't save when you changed nothing. This is because `imageUrls` isn't in this dependency array and since `useCallback` is employed, the initial value of the state (empty array) is used and isn't recomputed because `imageUrls` isn't in this array. It be should be added

**Review by BENJIGAO — CHANGES_REQUESTED** [review]

**Review by Samanthatb1 — CHANGES_REQUESTED** [review]
When I click the edit button, nothing happens. I believe the edit page is supposed to open but nothing opens on my end. Can you confirm this issue doesn't happen on your end with some of the new changes you made?

**Comment by klau1011** [comment]
> When I click the edit button, nothing happens. I believe the edit page is supposed to open but nothing opens on my end. Can you confirm this issue doesn't happen on your end with some of the new changes you made?

Hi - yeah it seems to works on my end. Perhaps you were trying to click 'edit description' on the current geese at the top instead of 'edit' on the geese in the table?

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:116]: A good idea might be to add a comment saying what this if statement computes (e.g., if this is a previous or new posting)

File: src/frontend/hooks/goose-form.js
Comment (BENJIGAO) [inline src/frontend/hooks/goose-form.js:140]: This line causes the bug of not being able to save the goose form for new gooses. This is due to using `imageUrls`, which is a map of the previous images, which in a new goose, is empty regardless of the images you add. I believe `newlyAddedImageUrls` should be used instead, which represents the new images. However, all in all, this naming convention of the `images` state representing new images and `imageUrls` representing previous images isn't great

**Review by BENJIGAO — CHANGES_REQUESTED** [review]
Other than a few small changes, lgtm! When you merge after the changes, do a squash and merge.

File: src/frontend/hooks/goose-form.js
Comment (Samanthatb1) [inline src/frontend/hooks/goose-form.js:140]: should this be `newlyAddedImageUrls`

Comment (klau1011) [reply src/frontend/hooks/goose-form.js:140]: > should this be `newlyAddedImageUrls`

whoops - good catch! - must've accidentally fat-fingered cntl-z'd before pushing.. :P

**Review by Samanthatb1 — APPROVED** [review]
Everything looks good to me. I tested the image upload in the PROD environment and they are uploaded correctly, nice job 👏🏻

**Review by BENJIGAO — APPROVED** [review]

**Pull request merged**
